### DISCUSSION TITLE
UTF-8 with BOM

### DISCUSSION THREAD

**Discussion by Lisible** [helix-editor/helix#4533] [Q&A] [answered]
Is there any way to add BOM (Byte-Order Mark) to a UTF-8 file from the editor ?
I encountered a situation where I needed to write a file with BOM and I haven't found a built-in way to do it.

If there is no current way to do it, I might do a PR for it if that feature is desired.

**Comment by CBenoit** [comment, marked as answer]
Indeed, I don’t think you can do that as of now. A simple enough feature that can come in handy however. PR welcomed 🙂 

EDIT: Some background for anyone interested: http://www.unicode.org/versions/Unicode5.0.0/ch02.pdf
![image](https://user-images.githubusercontent.com/3809077/199041803-ff830630-f712-4e56-a771-7db6d09b798d.png)
Not recommended and as such not default, but I imagine a new command for that. `:add-utf8-bom`?

**Reply by Lisible** [reply]
I was indeed thinking of a new command that would add it if not present. I'll try to find some time to work on that.

**Comment by archseer** [comment]
In `vim` this is a separate config setting called "bomb": https://vim.fandom.com/wiki/Working_with_Unicode
